### PR TITLE
Limit and prioritize Rclone transfers

### PR DESCRIPTION
**Pull request by KillerX**

### CODE REVIEW

File: services/rclone/queue.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		if count >= maxConcurrentTransfers {
+			return
+		}
+
+		for _, priority := range Priorities.Members() {
+			for _, ch := range transferQueue[priority] {
+				if count >= maxConcurrentTransfers {
+					return
+				}
+				ch <- true
```
Comment (andreasgangso) [inline services/rclone/queue.go:51]: add count++? otherwise the if statement above will never happen.

**Review by andreasgangso — CHANGES_REQUESTED** [review]

File: services/rclone/queue.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+func waitForTransferSlot(priority Priority, timeout time.Duration) (chan bool, error) {
+	ch := make(chan bool)
+
+	queueLock.Lock()
+	transferQueue[priority] = append(transferQueue[priority], ch)
+	queueLock.Unlock()
+
+	select {
+	case <-ch:
```
Comment (andreasgangso) [inline services/rclone/queue.go:28]: remove from the transferqueue when it's done?

File: services/rclone/queue.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	"time"
+)
+
+const maxConcurrentTransfers = 5
+
+var queueLock = sync.Mutex{}
+var transferQueue = map[Priority][]chan bool{}
+
+func init() {
+	for _, priority := range Priorities.Members() {
```
Comment (andreasgangso) [inline services/rclone/queue.go:15]: this `init` seems to never be called. But maybe it doesnt need to?

**Review by andreasgangso — CHANGES_REQUESTED** [review]

**Comment by KillerX** [comment]
Init is called when package is loaded.

File: services/rclone/queue.go (hunk 3 above)
Comment (KillerX) [reply services/rclone/queue.go:15]: Init is called when package is loaded.

**Review by andreasgangso — APPROVED** [review]
lgtm

**Comment by andreasgangso** [comment]
husk å slett branch 🤓

**Pull request merged**